JASP-Engine/JASP/R/correlationbayesian.R

```diff
@@ -15,2850 +15,1312 @@
-# }
-
-# 2.3 Savage-Dickey beta approximation
-.bfSavageDickeyBetaData <- function(n, r, kappa=1, rho0=0) {
-    # Savage-Dickey based on a beta approximation
+    # Calculate expected dimension
     #
+    if (nVariables <= 1) {
+      numberOfRows <- 1
+    } else {
+      numberOfRows <- choose(nVariables, 2)
+    }
+
+    if (is.null(methodItems) | length(methodItems)==0) {
+      numberOfColumns <- 3
+    } else {
+      numberOfColumns <- 3 + options[["reportN"]] + length(methodItems) * sum(options[["reportBayesFactors"]],
+                                                                              2*options[["ci"]])
+    }
+  } else {
+    # Note(Alexander): Correlation Matrix
     #
-    fit <- .posteriorBetaParameters(n=n, r=r, kappa=kappa)
-    return(.bfSavageDickeyBeta(a=fit$betaA, b=fit$betaB, kappa=kappa, rho0=rho0))
-}
+    corBayesTable$addColumnInfo(name="variable", title="", type="string")
+    corBayesTable$addColumnInfo(name="itemColumn", title="", type="number")
+
+    if (nVariables <= 1) {
+      if (nVariables == 0) {
+        corBayesTable$addColumnInfo(name="empty1", type="number", title="...")
+      } else {
+        corBayesTable$addColumnInfo(name=options[["variables"]][1], type="number", title=options[["variables"]][1])
+      }
+      corBayesTable$addColumnInfo(name="empty2", type="number", title="...")
+    } else {
+      for (variable in options[["variables"]]) {
+        corBayesTable$addColumnInfo(name=variable, type="number",title=variable)
+      }
+    }
+
+    if (length(methodItems)==0) {
+      nItems <- sum(options[["reportBayesFactors"]], 2*options[["ci"]], options[["reportN"]])
 
-.bfSavageDickeyOneSidedAdapt <- function(bf10, a, b, kappa=1, rho0=0) {
-    return(.bfSavageDickeyBeta(bf10=bf10, a=a, b=b, kappa))
+      if (nItems <= 1) {
+        numberOfColumns <- nVariables
+      } else {
+        numberOfColumns <- 2 + nVariables
+      }
+      numberOfRows <- nVariables
+    } else {
+      itemsPerMethod <- sum(options[["reportBayesFactors"]], 2*options[["ci"]])
+      nVariablesEffective <- max(nVariables, 2)
+
+      if (itemsPerMethod==0) {
+        numberOfColumns <- 1 + nVariablesEffective
+        numberOfRows <- nVariablesEffective
+      } else {
+        numberOfColumns <- 2 + nVariablesEffective
+        numberOfRows <- nVariablesEffective * length(methodItems) *
+          (itemsPerMethod + 1) + options[["reportN"]] * nVariablesEffective
+      }
+    }
+  }
+
+  # c. Assign ----
+  #
+  corBayesTable$setExpectedSize(numberOfRows, numberOfColumns)
+  jaspResults[["corBayesTable"]] <- corBayesTable
+
+  # d. Compute ----
+  #
+  allBfObjects <- .computeCorBayes("allBfObjects"=allBfObjects, "dataset"=dataset,
+                                   "options"=options, "computePlots"=FALSE,
+                                   "purpose"="matrix", "ready"=ready)
+
+  # e. Fill ----
+  #
+  allBfObjects <- .fillCorBayesTable("table"=corBayesTable, "options"=options, 
```

Comment:
Same here, what does the filling of a table have to do with this object?
Here I'd expect a table to be returned or nothing (because of pass by reference of the table)